.clang-format

```diff
@@ -0,0 +1,96 @@
+---
```

Comment:
the file looks big with unnecessary stuff, or put some reference so reviewer can cross reference.

Reply:
Alright, I will put a reference to the documentation.